OnlineStore/OnlineStore.WebApi/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using OnlineStore.BusinessLogic.Interfaces;
+using OnlineStore.DTO.DTO;
+
+namespace OnlineStore.WebApi.Controllers
+{
+    [ApiController]
+    [Route("category")]
+    public class CategoryController : ControllerBase
+    {
+        private readonly ICategoryService _categoryService;
+
+        public CategoryController(ICategoryService categoryService)
+        {
+            _categoryService = categoryService ?? throw new NullReferenceException(nameof(categoryService));
+        }
+
+        //[Authorize(Roles = "Admin")]
+        [HttpPatch]
+        public async Task<IActionResult> UpdateCategory([FromBody] CategoryDto categoryDto)
+        {
+            await _categoryService
+                    .UpdateCategoryAsync(categoryDto, CancellationToken.None);
+
+            return Ok();
+        }
+
+        //[Authorize(Roles = "Admin")]
+        [HttpDelete("{id:int}")]
+        public async Task<IActionResult> DeleteCategory(int id)
```

Comment:
If you specify attributes, add them for data from the routing as well to maintain a consistent style